### PR TITLE
Issue 7273 geth style genesis

### LINKED ISSUE #7273: Allowing Besu to parse a Geth-style genesis like other clients
### Description
As a node operator, I would like Besu to be able to parse configs in the same way as other clients, so interoperability testing and multi-client network operation is simpler.

### PR DESCRIPTION
**Pull request by cloudspores**
## Fixed Issue(s)
Fixes #7273 
Allowing Besu to parse a Geth-style genesis like other clients, so interoperability testing and multi-client network operation is simpler.

### Thanks for sending a pull request! Have you done the following?

- [x] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [x] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [x] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [x] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [x] unit tests: `./gradlew build`
- [x] acceptance tests: `./gradlew acceptanceTest`
- [x] integration tests: `./gradlew integrationTest`
- [x] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

**Review by fab-10 — CHANGES_REQUESTED** [review]
Overall looks good, there are some minor things to remove or change.
For the tests, I suggest to add also some real world geth genesis files, for example from the current [Pectra devnets](https://github.com/ethpandaops/pectra-devnets/tree/master/network-configs/devnet-3/metadata), where `genesis.json` is in geth format and `besu.json` is in Besu

File: config/src/main/java/org/hyperledger/besu/config/GenesisFileAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+    // Check for Geth-specific extraData format that Besu doesn't support
+    if (genesisJson.has("extradata")) {
+      String extraData = genesisJson.get("extradata").asText();
+      if (extraData != null && !extraData.isEmpty()) {
+        if (extraData.length() == 64 || extraData.length() == 32) {
+          fileStyle = GenesisAnalysis.FileStyle.GETH;
+        }
+      }
+    }
```
[the remark this answers is not in the document]
Comment (cloudspores) [reply config/src/main/java/org/hyperledger/besu/config/GenesisFileAnalyzer.java]: I did more research of the extraData formatting and as far as I can see, the Geth-style formatting of extraData for Clique is the same as for Besu-style.

File: gradle/verification-metadata.xml [hunk 2]
```diff
@@ -3512,6 +3520,22 @@
...
+         </artifact>
+      </component>
+      <component group="org.apache.tuweni" name="tuweni-units" version="2.3.1">
+         <artifact name="tuweni-units-2.3.1.jar">
+            <sha256 value="59703a35883555ec944149aca53c81c26b2b9200520ffde059d57326d2ec7ca4" origin="Generated by Gradle"/>
+         </artifact>
+         <artifact name="tuweni-units-2.3.1.pom">
+            <sha256 value="51b353af00ee41095bacae9719c39e7c5e5b37ab5c4e405eebfe8c7056db7077" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
```
[the remark this answers is not in the document]
Comment (cloudspores) [reply gradle/verification-metadata.xml:3538]: I have reverted these changes, thanks.

**Review by fab-10 — COMMENTED** [review]
Nice to have the test against real world genesis files.

Before approving I would like to understand potential performance penalty of the extra parsing, see my comment https://github.com/hyperledger/besu/issues/7273#issuecomment-2368714262

File: CHANGELOG.md [hunk 3]
```diff
@@ -20,6 +20,10 @@
 
 ### Upcoming Breaking Changes
 
+### Additions and Improvements
+- Allowing Besu to parse a Geth-style genesis like other clients [#7273](https://github.com/hyperledger/besu/pull/7273)
+- Let Besu parse configs in the same way as other clients, so interoperability testing and multi-client network operation is simpler
```
[the remark this answers is not in the document]
Comment (cloudspores) [reply CHANGELOG.md:25]: Fixed, I thought was referring to the check in command.

**Comment by fab-10** [comment]
I can take care of this one, evaluating the performance impact

**Pull request closed without merging**